ion-schema/src/isl/version_based_isl/constraints/all_of.rs

```diff
@@ -0,0 +1,16 @@
+use crate::isl::version_based_isl::{IslTypeRef, IslVersionTrait};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct AllOf<V: IslVersionTrait> {
```

Comment:
🗺️ All the constraints are now placed inside `constraints` module. Each constraint will now accept a generic `V` that represents the ISL version using version marker trait.